src/inabox/inabox-viewport.js

```diff
@@ -467,6 +467,13 @@ export class ViewportBindingInabox {
     /** @private {?UnlistenDef} */
     this.unobserveFunction_ = null;
 
+    const unloadListener = this.win.addEventListener('unload', () => {
```

Comment:
is it guaranteed to fire `visibilitychange` on window `unload`?

if so, we might want to fix the `visibilitychange` regression. and we will need to make sure inaboxViewportBinding has idempotent `connect()` `disconnect()`.

Reply:
It looks like the case: https://w3c.github.io/page-visibility/ (though [Safari might behave differently](https://bugs.webkit.org/show_bug.cgi?id=151234)?

Making `visibilityChange` fire again does seem like the solution. I'll fix it, test it on Safari and get back to you once it's done.

Reply:
Unfortunately from testing looks like Safari does not fire `visibilityChange` on `unload`, so this fix doesn't work there.

Reply:
The next thing to try is probably the `pagehide` event, which I think is acceptable?

Reply:
SGTM.

does that mean the issue existed in Safari before inabox-lite?